src/porepy/applications/test_utils/common_xpfa_tests.py

```diff
@@ -818,3 +818,118 @@ def test_linear_pressure_dirichlet_conditions_perturbed_grid(self):
+    elif isinstance(discr_class, pp.Biot):
+        # Both keywords are set for Biot. We need to deal with this before the Mpsa
+        # case, since Biot is a subclass of Mpsa.
+        flow_keyword = discr_class.flow_keyword
+        mechanics_keyword = discr_class.mechanics_keyword
+    elif isinstance(discr_class, pp.Mpsa):
+        # Pick the keyword set for the mechanics problem, set a default for the flow.
+        flow_keyword = "flow"
+        mechanics_keyword = discr_class.keyword
+
+    # Consider four different grids; this should be enough to cover all cases. The size
+    # of the grids are a bit random here, what we really want is to allow a division
+    # into subgrids that include an overlap, but also cells on each side which is not
+    # part of the overlap.
+    #
+    # Note: While it is tempting to use parametrization for this loop over grids, EK
+    # found no easy way to do that which also allows for calling this test from a
+    # shallow test wrapper which is unique to each discretization class (we could have
+    # used the same parametrization on each of the wrappers, but that would have been a
+    # bit messy).
+    grid_list = [
+        pp.CartGrid(np.array([4, 2])),
+        pp.CartGrid(np.array([4, 2, 2])),
+        pp.StructuredTriangleGrid(np.array([4, 2])),
+        pp.StructuredTetrahedralGrid(np.array([4, 2, 2])),
+    ]
+
+    # Loop over the grids, discretize twice (with different data dictionaries): Once
+    # wiht an enforced splitting into subdomains, once with no such splitting.
+    for g in grid_list:
+        g.compute_geometry()
+        nc = g.num_cells
+
+        # Parameter dictionaries for the two types of physics, can be common for both
+        # the partitioned and non-partitioned schemes. Give boundary conditions and
+        # tensors for both types of physics; although we may only use one of them
+        # (unless Biot), but the overhead should be minimal.
+        flow_param = {
+            "bc": pp.BoundaryCondition(g),
+            "second_order_tensor": pp.SecondOrderTensor(np.ones(nc)),
+            "partition_arguments": {"num_subproblems": 2},
+        }
+        mechanics_param = {
+            "bc": pp.BoundaryConditionVectorial(g),
+            "fourth_order_tensor": pp.FourthOrderTensor(np.ones(nc), np.ones(nc)),
+            "partition_arguments": {"num_subproblems": 2},
+            "biot_alpha": 1,
+        }
+
+        # Set up a data dictionary that will split the discretization into two.
+        data_partition = {
+            pp.PARAMETERS: {
+                flow_keyword: flow_param,
+                mechanics_keyword: mechanics_param,
+            },
+            pp.DISCRETIZATION_MATRICES: {flow_keyword: {}, mechanics_keyword: {}},
+        }
+        # Discretize
+        discr_class.discretize(g, data_partition)
+
+        # Set up a data dictionary that will not split the discretization into two.
+        data_no_partition = {
```

Comment:
Doesn't this reuse `"partition_arguments": {"num_subproblems": 2},` from above and hence partition in two subgrids?

Reply:
Good point. Thanks.